### PR TITLE
Login/Registration Page Integration with backend

### PR DESCRIPTION
**Pull request by evankp**
- [x] Register page API
- [x] Login page API
- [x] Login with email not name
- [x] Error handling
- [x] JWT Token storage
- [x] Modified User model so a users name is not unique, IE authenticate by id or email

### CODE REVIEW

**Comment by evankp** [comment]
Working on a toast for successful login, but due to it not being needed to function, I am going to move it to a separate PR

File: client/src/pages/Login.js [hunk 1]
```diff
@@ -1,150 +1,158 @@
...
+      const { email, password } = this.state;
+      const userLogin = await authenticateUser('login', { email, password });
+
+      if (userLogin.hasOwnProperty('err')) {
+        validation[userLogin.property] = {
+          isInvalid: true,
+          message: capitalize(userLogin.err)
+        };
+        return this.setState({ validation });
+      }
```
Comment (evankp) [inline client/src/pages/Login.js:104]: Due to the Prettier formatting changes, I changed the most around here

File: client/src/helpers/UserHelpers.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import axios from 'axios';
```
Comment (tsyaeger) [inline client/src/helpers/UserHelpers.js]: We can also use an axios config to do some of this work. Maybe we chat about somehow fusing this and my api in the profile page?

File: client/src/pages/SignUp.js [hunk 3]
```diff
@@ -1,182 +1,190 @@
...
+        validation[userRegistration.property] = {
+          isInvalid: true,
+          message: capitalize(userRegistration.err)
+        };
+        return this.setState({ validation });
+      }
+
+      if (userRegistration.hasOwnProperty('success')) this.props.history.push('/profile');
+    }
+  };
```
Comment (evankp) [inline client/src/pages/SignUp.js]: Due to the Prettier formatting changes, I changed the most around here

**Comment by evankp** [comment]
Yeah maybe we can work on a little bit after stand-up today

File: client/src/helpers/UserHelpers.js (hunk 2 above)
Comment (DevDelvo) [reply client/src/helpers/UserHelpers.js:1]: That would be great! And I'd need some help understanding it as well

Comment (evankp) [reply client/src/helpers/UserHelpers.js:1]: Yes that's be cool

File: client/src/pages/Login.js [hunk 4]
```diff
@@ -1,150 +1,158 @@
 import React from 'react';
-import {
-    withStyles,
-    Button
-} from '@material-ui/core';
-import {Link} from 'react-router-dom';
+import { withStyles, Button } from '@material-ui/core';
+import { capitalize } from '@material-ui/core/utils';
+import { Link } from 'react-router-dom';
```
Comment (DevDelvo) [inline client/src/pages/Login.js:4]: we might have to come to an agreement on more coding styles?

File: client/src/helpers/UserHelpers.js (hunk 2 above)
Comment (evankp) [reply client/src/helpers/UserHelpers.js:1]: I'll think about it and bring it up at standup

File: client/src/pages/Login.js (hunk 4 above)
Comment (evankp) [reply client/src/pages/Login.js:4]: Hmm this was done by prettier

File: client/src/helpers/UserHelpers.js (hunk 2 above)
Comment (evankp) [reply client/src/helpers/UserHelpers.js:1]: Decided we will do this in @tsyaeger PR

**Comment by evankp** [comment]
I'll let @DevDelvo merge this PR if approved and I am busy so you two can get started on conflicts and such while I am away for the evening shift 👍

**Review by DevDelvo — APPROVED** [review]

**Pull request merged**